src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,53 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int GROWTH_FACTOR = 2;
+    private T[] stack;
+    private int size;
+    private int capacity;
 
+    public MateStack() {
+        final int default_capacity = 10;
```

Comment:
The variable `default_capacity` should follow Java naming conventions for constants with all uppercase letters and underscores, e.g., `DEFAULT_CAPACITY`.